hornet-private-net/extra-nodes/private-hornet.sh

```diff
@@ -74,12 +64,17 @@ else
 fi
 
 if [ -n "$4" ]; then
-  peer_address="$4"
-else 
-  if [ -f ../node1.identity.txt ]; then
-    peer_address="/dns/node1/tcp/15600/p2p/$(getPeerID ../node1.identity.txt)"
+  # We determine whether the address is a peer address or an entry node address
+  if [ "$4" =~ .*"\/autopeering\/".* ]; then
```

Comment:
Running the command
```
./private-hornet.sh install  node004 8b54ebd10a02f6201f226fac78ca030d4f73f7ed10022d085d9ad8eca79906a0 "/dns/spammer/tcp/15600/p2p/12D3KooWBnxutsTN2yYPkipDbLZq3kvb44PXG97ry4UPbnpE74wX"
```
It deploys the node successfully. However I get the error

```
./private-hornet.sh: line 68: [: =~: binary operator expected
```
From
```
Error response from daemon: network with name private-tangle already exists
Setting permissions for Hornet...
node004
Creating node004_node004_run ... done
Setting up peer node: /dns/spammer/tcp/15600/p2p/12D3KooWBnxutsTN2yYPkipDbLZq3kvb44PXG97ry4UPbnpE74wX
Creating node004 ... done
```

The code works when adding a regular peer. But when it comes to autopeering

```
./private-hornet.sh install  node005 8b54ebd10a02f6201f226fac78ca030d4f73f7ed10022d085d9ad8eca79906a0 "/dns/node-autopeering/udp/14626/autopeering/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ"
```
I get this
```
./private-hornet.sh: line 68: [: =~: binary operator expected

```
And it tries to add an autopeering  string as a regular peer node

```
Setting up peer node: /dns/node-autopeering/udp/14626/autopeering/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ
```
From 

```
./private-hornet.sh: line 68: [: =~: binary operator expected
Error response from daemon: network with name private-tangle already exists
Setting permissions for Hornet...
Creating node005_node005_run ... done
Setting up peer node: /dns/node-autopeering/udp/14626/autopeering/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ
Creating node005 ... done
```
If I try adding back ticks, the problem is still there

```
[ec2-user@ip-172-31-45-25 extra-nodes]$ ./private-hornet.sh install  node005 8b54ebd10a02f6201f226fac78ca030d4f73f7ed10022d085d9ad8eca79906a0 "\/dns\/node-autopeering\/udp\/14626\/autopeering\/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXg
N6gaZ"
./private-hornet.sh: line 68: [: =~: binary operator expected
Error response from daemon: network with name private-tangle already exists
Setting permissions for Hornet...
node005
Creating node005_node005_run ... done
Setting up peer node: \/dns\/node-autopeering\/udp\/14626\/autopeering\/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ
Creating node005 ... done
```

Reply:
Adding back ticks on  peering a regular node works. Even when I go and check the logs.
```
 ./private-hornet.sh install  node004 8b54ebd10a02f6201f226fac78ca030d4f73f7ed10022d085d9ad8eca79906a0 "\/dns\/spammer\/tcp\/15600\/p2p\/12D3KooWBnxutsTN2yYPkipDbLZq3kvb44PXG97ry4UPbnpE74wX"
./private-hornet.sh: line 68: [: =~: binary operator expected
Error response from daemon: network with name private-tangle already exists
Setting permissions for Hornet...
node004
Creating node004_node004_run ... done
Setting up peer node: \/dns\/spammer\/tcp\/15600\/p2p\/12D3KooWBnxutsTN2yYPkipDbLZq3kvb44PXG97ry4UPbnpE74wX
Creating node004 ... done
```

When adding double braces around the line
```
if [[ "$4" =~ .*"\/autopeering\/".* ]]; then
```
Autopeering works with the following format. it detects it's an autopeering address.

```
 ./private-hornet.sh install  node005 8b54ebd10a02f6201f226fac78ca030d4f73f7ed10022d085d9ad8eca79906a0 "\/dns\/node-autopeering\/udp\/14626\/autopeering\/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ"
Error response from daemon: network with name private-tangle already exists
Setting permissions for Hornet...
node005
Creating node005_node005_run ... done
Setting up autopeering entry node: \/dns\/node-autopeering\/udp\/14626\/autopeering\/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ
Creating node005 ... done
```

Without the back ticks, it doesn't seem to work for autopeering, as it is taken up as a peer node instead of an autopeering node

```
./private-hornet.sh install  node005 8b54ebd10a02f6201f226fac78ca030d4f73f7ed10022d085d9ad8eca79906a0 "/dns/node-autopeering/udp/14626/autopeering/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ"
Error response from daemon: network with name private-tangle already exists
Setting permissions for Hornet...
node005
Creating node005_node005_run ... done
Setting up peer node: /dns/node-autopeering/udp/14626/autopeering/12D3KooWGGmSgfyGvWb6reLrsTbtJKUsU323ye6E1PvVFXgN6gaZ
Creating node005 ... done
```